src/reconcile/rec_visibility.c

```diff
@@ -726,6 +742,17 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
         supd_restore = F_ISSET(r, WT_REC_EVICT) &&
           (has_newer_updates || F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
 
+        /*
+         * The total update size only contains uncommitted updates. This is wrong for the in memory
+         * case because we cannot discard any update until they are obsolete. Add them to the size.
```

Comment:
Not adding the updates to the size doesn't discard them right? May be updating the comment as they are not considered in in-memory split calculation?

Reply:
They will be discarded after the reconciliation. The onpage value and the values moved to history store will be freed.

Reply:
But this condition is for the in-memory scenario, where we do not free the updates. My main point is that from the comment, It may be misunderstand that the below code can avoid freeing the updates, but it doesn't.